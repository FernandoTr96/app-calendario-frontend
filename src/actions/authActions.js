import { fetchWithoutToken, fetchWithToken } from "../helpers/fetch";
import { types } from "../types/types";
import { toast } from 'react-toastify';

export const startLoginWithEmailAndPassword = (email, password)=>{

    return async (dispatch)=>{

        const response = await fetchWithoutToken(
            'auth',
            {email, password},
            'POST'
        );

        const body = await response.json();
        
        if(body.ok){
            localStorage.setItem('calendarAppToken', body.token );
            localStorage.setItem('token-init-date', new Date().getTime() );
            delete body.token;
            delete body.ok;
            dispatch(login(body));
        }
        else{

            toast.error('usuario o contraseÃ±a incorrecta', {
                position: "top-right",
                autoClose: 5000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: false,
                draggable: false,
                progress: undefined,
            });

        }

    }

}

export const login = (userCredentials)=>({
    type: types.login,
    payload: userCredentials
})

export const registerAccount = (userdata)=>{
    
    return async (dispatch)=>{

        const response = await fetchWithoutToken(
            'auth/register',
            {...userdata},
            'POST'
        );

        const body = await response.json();

        if(body.ok){
            localStorage.setItem('calendarAppToken', body.token );
            localStorage.setItem('token-init-date', new Date().getTime() );
            delete body.token;
            delete body.ok;
            delete body.confirm_password;
            dispatch(login(body));
        }
        else{

            toast.error(body.msg, {
                position: "top-right",
                autoClose: 5000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: false,
                draggable: false,
                progress: undefined,
            });

        }

        if(body.errors?.length > 0){
            body.errors.forEach(field => {

                toast.error(field.msg, {
                    position: "top-right",
                    autoClose: 5000,
                    hideProgressBar: false,
                    closeOnClick: true,
                    pauseOnHover: false,
                    draggable: false,
                    progress: undefined,
                });

            })
        }

    }

}

export const startChecking = ()=> {

    return async (dispatch)=>{

        const response = await fetchWithToken(
            'auth/revalidate-token',
            'GET'
        );

        const body = await response.json();

        if(body.ok){
            localStorage.setItem('calendarAppToken', body.token );
            localStorage.setItem('token-init-date', new Date().getTime() );
            delete body.token;
            delete body.ok;
            dispatch(login(body));
        }
        else{
            dispatch(checkingFinish());
        }

    }

}

export const checkingFinish = ()=>({
    type: types.checkingFinish
})

export const startLogout = ()=>{

    return (dispatch)=>{

        localStorage.clear();
        dispatch(logout());

    }

}

export const logout = ()=>({
    type: types.purgeAuth
})