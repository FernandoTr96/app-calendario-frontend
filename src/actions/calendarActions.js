import { types } from "../types/types";
import {fetchWithToken} from '../helpers/fetch'
import {toast} from 'react-toastify'
import { string2date } from "../helpers/string2date";

export const getEvents = ()=>{
    return async (dispatch)=>{

        dispatch(showLoader());

        try {

            const response = await fetchWithToken('events');
            const body = await response.json();
            

            if(response.ok){
                
                const events = string2date(body.events);
                dispatch(setEvents(events));

            }

            
        } catch (error) {

            toast.error(`Error : ${error}`, {
                position: "bottom-right",
                autoClose: 4000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: false,
                draggable: false,
                progress: undefined,
            });

        }

        dispatch(hiddeLoader());

    }
}

export const setEvents = (events) =>
(
    {
        type: types.fillEventList,
        payload: events
    }
)

export const saveEvent = (event)=>{
    return async (dispatch) => {

        dispatch(showLoader());

        try {

            const insert = await fetchWithToken(
                'events/addEvent',
                event,
                'POST'
            );

            if(insert.ok){
                toast.success(`Se ha aÃ±adido un nuevo evento al calendario`, {
                    position: "bottom-right",
                    autoClose: 3000,
                    hideProgressBar: false,
                    closeOnClick: true,
                    pauseOnHover: false,
                    draggable: false,
                    progress: undefined,
                });
                dispatch(getEvents());
            }
            
        } catch (error) {
            
            toast.error(error.msg, {
                position: "bottom-right",
                autoClose: 5000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: false,
                draggable: false,
                progress: undefined,
            });

        }

        dispatch(hiddeLoader());

    }
}


export const editEvent = (event)=>{
    return async (dispatch)=>{

        const id = event.id;
        delete event.id;

        dispatch(showLoader());

        try {

            const update =  await fetchWithToken(
                `events/update/${id}`,
                event,
                'PUT'
            );

            if(update.ok){

                toast.info(`evento actualizado con exito`, {
                    position: "bottom-right",
                    autoClose: 3000,
                    hideProgressBar: false,
                    closeOnClick: true,
                    pauseOnHover: false,
                    draggable: false,
                    progress: undefined,
                });

                dispatch(getEvents());

            }

        } catch (error) {
            
            toast.error(`Error: ${error.msg}`, {
                position: "bottom-right",
                autoClose: 5000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: false,
                draggable: false,
                progress: undefined,
            });
        }

        dispatch(hiddeLoader());

    }
}

export const deleteEvent = (id)=>{
    return async (dispatch)=>{

        dispatch(showLoader());

        try {

            const _delete = await fetchWithToken(
                `events/delete/${id}`,
                {},
                'DELETE'
            )

            if(_delete.ok){

                toast.warning(`el evento fue eliminado con exito`, {
                    position: "bottom-right",
                    autoClose: 3000,
                    hideProgressBar: false,
                    closeOnClick: true,
                    pauseOnHover: false,
                    draggable: false,
                    progress: undefined,
                });

                dispatch(getEvents());
            }

            
        } catch (error) {

            toast.error(`Error: ${error.msg}`, {
                position: "bottom-right",
                autoClose: 5000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: false,
                draggable: false,
                progress: undefined,
            });

        }

        dispatch(hiddeLoader());

    }
}

export const activeEvent = (event)=>(
    {
        type: types.activeEvent,
        payload: event
    }
)

export const showLoader = ()=>(
    {
        type: types.showLoader
    }
)
export const hiddeLoader = ()=>(
    {
        type: types.hiddeLoader
    }
)