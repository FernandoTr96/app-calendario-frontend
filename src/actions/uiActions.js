import { types } from "../types/types";

export const openModalEvents = ()=>(
    {
        type: types.uiOpenModalEvents,
        payload: true
    }
);

export const closeModalEvents = ()=>(
    {
        type: types.uiCloseModalEvents,
        payload: false
    }
);