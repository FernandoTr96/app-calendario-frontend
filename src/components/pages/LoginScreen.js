import React from 'react'
import { Box } from '@mui/material'
import styles from '../../styles/mui-sx-styles/LoginScreenStyles'
import { SliderControls } from '../auth/SliderControls'
import { LoginForm } from '../auth/LoginForm'
import { RegisterForm } from '../auth/RegisterForm'
import { ToastContainer } from 'react-toastify';

export const LoginScreen = () => {
    return (

        // main container
        <Box component='main' sx={styles.mainContainer} >

            {/* auth container */}
            <Box component='div' sx={styles.authContainer} >

                {/* tabs */}
                <SliderControls />

                {/* slider */}
                <Box component='div' sx={styles.slider} className='slider'>

                    {/* LOGIN */}
                    <LoginForm />

                    {/* REGISTER */}
                    <RegisterForm />

                </Box>

            </Box>

            <ToastContainer
                position="top-right"
                autoClose={5000}
                hideProgressBar={false}
                newestOnTop={false}
                closeOnClick
                rtl={false}
                pauseOnFocusLoss={false}
                draggable={false}
                pauseOnHover={false}
            />

        </Box>

    )
}
