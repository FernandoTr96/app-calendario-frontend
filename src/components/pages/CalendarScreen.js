import React, { useEffect } from 'react'
import { NavBar } from '../ui/NavBar'
import { Box } from '@mui/material'

// importaciones de big calendar 
// https://github.com/jquense/react-big-calendar#readme// 
import { Calendar, momentLocalizer } from 'react-big-calendar'
import moment from 'moment'
import 'moment/locale/es-mx';
import {messages} from '../../helpers/calendar-messages'
import 'react-big-calendar/lib/css/react-big-calendar.css'
import { theme } from '../../styles/mui-core-theme/materialui'
import { ModalEvents } from '../ui/ModalEvents'
import { useSelector, useDispatch } from 'react-redux'
import { activeEvent, getEvents } from '../../actions/calendarActions'
import { ToastContainer } from 'react-toastify';
import Stack from '@mui/material/Stack';
import LinearProgress from '@mui/material/LinearProgress';


export const CalendarScreen = () => {

    const dispatch = useDispatch();
    const {eventList, loader} = useSelector(state => state.events);
    const {uid} = useSelector(state => state.auth.account);

    const localizer = momentLocalizer(moment);
    const eventStyleGetter = (event /*, start, end, isSelected*/)=>{
        
        const style = {
            backgroundColor: event.user._id === uid ? theme.palette.secondary.light : theme.palette.secondary.dark,
            borderRadius: 'none',
            display: 'block',
            color: 'wintersmoke',
            outlineColor: '#333'
        }

        return{
            style
        }

    };

    const onSelectEvent = (eventObj) =>{
        dispatch(activeEvent(eventObj));
    }

    const onSelectSlot = ()=>{
        dispatch(activeEvent(null));
    }

    useEffect(() => {

        dispatch(getEvents());

    }, [dispatch])

    return (
        <Box component='main' className='calendar-screen'>
            
            {
                loader && 
                <Stack sx={{ width: '100%', color: 'grey.500' }} spacing={2}>
                    <LinearProgress color="warning" />
                </Stack>
            }

            <NavBar />

            <Calendar
                localizer={localizer}
                events={eventList}
                startAccessor='start'
                endAccessor='end'
                messages={messages}
                eventPropGetter={eventStyleGetter}

                onSelectEvent={onSelectEvent}
                onSelectSlot={onSelectSlot}
                selectable={true}
            />

            <ModalEvents/>

            <ToastContainer
                position="top-right"
                autoClose={5000}
                hideProgressBar={false}
                newestOnTop={false}
                closeOnClick
                rtl={false}
                pauseOnFocusLoss={false}
                draggable={false}
                pauseOnHover={false}
            />
        </Box>
    )
}
