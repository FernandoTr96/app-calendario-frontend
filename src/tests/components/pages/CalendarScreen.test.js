import {mount} from 'enzyme'
import thunk from 'redux-thunk'
import configure from 'redux-mock-store'
import {Provider} from 'react-redux'
import {MemoryRouter} from 'react-router-dom'
import { CalendarScreen } from '../../../components/pages/CalendarScreen';
import {activeEvent, getEvents} from '../../../actions/calendarActions'

const middlewares = [thunk];
const mockstore = configure(middlewares);
const initialState = {
    events:{
        eventList: [],
        loader: false,
        activeNow: null
    },
    auth:{
        account:{
            uid: 'abc12346'
        }
    },
    ui:{
        modalEventsOpen: false
    }
};
let store = mockstore(initialState);
store.dispatch = jest.fn();

let wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <CalendarScreen/>
        </MemoryRouter>
    </Provider>
);

jest.mock('../../../actions/calendarActions',()=>({
    activeEvent: jest.fn(),
    getEvents: jest.fn()
}))

describe('Pruebas en <CalendarScreen/>', () => {

    beforeEach(()=>{
        wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <CalendarScreen/>
                </MemoryRouter>
            </Provider>
        );
    })
    
    test('debe renderizarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('debe disparar la accion activeEvent al seleccionar el evento del calendario o el slot', () => {
        
        const calendar = wrapper.find('Calendar');
        const event = calendar.prop('onSelectEvent');
        const slot = calendar.prop('onSelectSlot');

        event();
        slot();

        expect(activeEvent).toHaveBeenCalled();
        expect(getEvents).toHaveBeenCalled();

    });

});