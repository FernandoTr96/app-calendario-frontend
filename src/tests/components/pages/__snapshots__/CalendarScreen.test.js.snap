// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CalendarScreen/> debe renderizarse correctamente 1`] = `
<main
  className="calendar-screen MuiBox-root css-0"
>
  <div
    className="MuiBox-root css-i9gxme"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <div
          className="css-e53awj-MuiStack-root"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-c6b2m7-MuiAvatar-root"
          >
            u+
          </div>
        </div>
        <span
          className="MuiTypography-root MuiTypography-span css-1w0zw7x-MuiTypography-root"
        />
        Array [
          <span
            aria-label="addEvent"
            aria-labelledby={null}
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="EventIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
              />
            </svg>
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>,
          "",
        ]
        Array [
          <span
            aria-label="logout"
            aria-labelledby={null}
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="LogoutIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
              />
            </svg>
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>,
          "",
        ]
      </div>
    </header>
  </div>
  <div
    className="rbc-calendar"
  >
    <div
      className="rbc-toolbar"
    >
      <span
        className="rbc-btn-group"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Hoy
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          «
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          »
        </button>
      </span>
      <span
        className="rbc-toolbar-label"
      >
        noviembre 2021
      </span>
      <span
        className="rbc-btn-group"
      >
        <button
          className="rbc-active"
          key="month"
          onClick={[Function]}
          type="button"
        >
          Mes
        </button>
        <button
          className=""
          key="week"
          onClick={[Function]}
          type="button"
        >
          Semana
        </button>
        <button
          className=""
          key="day"
          onClick={[Function]}
          type="button"
        >
          Día
        </button>
        <button
          className=""
          key="agenda"
          onClick={[Function]}
          type="button"
        >
          Agenda
        </button>
      </span>
    </div>
    <div
      aria-label="Month View"
      className="rbc-month-view"
      role="table"
    >
      <div
        className="rbc-row rbc-month-header"
        role="row"
      >
        <div
          className="rbc-header"
          key="header_0"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            dom.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_1"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            lun.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_2"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            mar.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_3"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            mié.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_4"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            jue.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_5"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            vie.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_6"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            sáb.
          </span>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                31
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                01
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                02
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                03
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                04
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                05
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                06
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                07
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                08
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                09
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                10
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                11
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                12
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                13
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg rbc-today"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                14
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                15
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                16
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-now rbc-current"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                17
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                18
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                19
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                20
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                21
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                22
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                23
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                24
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                25
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                26
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                27
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                28
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                29
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                30
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                01
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                02
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                03
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                04
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Toastify"
  />
</main>
`;
